include/spblas/detail/view_inspectors.hpp

```diff
@@ -83,23 +83,25 @@ bool has_scaling_factor(T&& t) {
 
 template <tensor T>
 auto get_ultimate_base(T&& t) {
-    if constexpr (has_base<T>) {
-        if constexpr (is_matrix_opt_view_v<T>) {
-            return t;
-        } else { 
-            return get_ultimate_base(t.base());
-        }
+  if constexpr (has_base<T>) {
+    if constexpr (is_matrix_opt_view_v<T>) {
+      return t;
     } else {
-        return t;
+      return get_ultimate_base(t.base());
     }
+  } else {
+    return t;
+  }
```

Comment:
so get_ultimate_base still returns not the ultimate base but either a matrix_opt or the bottom xyz_view now ?  do we want to keep that as is ?  or should we return it back to ignoring matrix_opt_view_v<T> ?